boostedblob/azure_auth.py

```diff
@@ -142,6 +146,17 @@ async def get_access_token(cache_key: Tuple[str, Optional[str]]) -> Tuple[Any, f
     now = time.time()
     creds = load_credentials()
 
+    # If opted into using azure-identity, use DefaultAzureCredential to get a token
+    # This enables the use of Managed Identity, Workload Identity, and other auth methods not implemented here
+    if creds["_azure_auth"] == "azure-identity":
+        from azure.identity import DefaultAzureCredential
+
+        cred = DefaultAzureCredential()
+        token = cred.get_token("https://storage.azure.com/.default")
```

Comment:
Doesn't azure-identity have an asyncio subpackage ? Would be good to avoid blocking event loop